crates/noirc_evaluator/src/ssa/context.rs

```diff
@@ -226,10 +267,32 @@ impl<'a> SsaContext<'a> {
         opcode: node::Operation,
         optype: node::ObjectType,
     ) -> NodeId {
+        let mut operands = vec![lhs, rhs];
+        self.new_instruction_with_multiple_operands(&mut operands, opcode, optype)
+    }
+
+    pub fn new_instruction_with_multiple_operands(
+        &mut self,
+        operands: &mut Vec<NodeId>,
+        opcode: node::Operation,
+        optype: node::ObjectType,
+    ) -> NodeId {
+        while operands.len() < 2 {
+            operands.push(NodeId::dummy());
+        }
         //Add a new instruction to the nodes arena
-        let mut i = node::Instruction::new(opcode, lhs, rhs, optype, Some(self.current_block));
+        let mut i = node::Instruction::new(
+            opcode,
+            operands[0],
+            operands[1],
+            optype,
+            Some(self.current_block),
+        );
         //Basic simplification
         optim::simplify(self, &mut i);
+        if operands.len() > 2 {
+            i.ins_arguments = operands.clone();
```

Comment:
So the assumption was that all instructions have rhs and lhs.
Now that we need more than 2 arguments for function calls, they all are put inside this ins_arguments field, but rhs and lsh are still used for all the other cases so it is safer to use them also here. lhs and rhs can now be abstracted away but this would require to refactor a bunch of code.

Reply:
Ok, I think I'll do a refactoring pass sometime then. Since the number of arguments is dependent on the opcode, I'd like the opcode to be a tagged union